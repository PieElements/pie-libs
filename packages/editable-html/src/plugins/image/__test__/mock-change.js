import { Data } from 'slate';
import { stub } from 'sinon';

export default function MockChange() {

  this.setNodeByKey = stub().returns(this);

  this.removeNodeByKey = stub().returns(this);
  this.insertInline = stub().returns(this);
}

export function MockDocument() {

  this.getChild = stub().returns({ data: Data.create({}) });

  this.getDescendant = stub();
}