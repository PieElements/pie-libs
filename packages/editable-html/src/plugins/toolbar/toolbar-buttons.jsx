import React from 'react';
import debug from 'debug';
import injectSheet from 'react-jss';

const styles = {
  button: {
    color: 'grey',
    display: 'inline-flex',
    padding: '2px',
    '& :hover': {
      color: 'black'
    }
  },
  active: {
    color: 'black'
  }
}

const buttonStyles = {
  root: Object.assign(styles.button, {
    display: 'inline-flex'
  })
}

const rbLog = debug('editable-html:raw-button');

export class RawButton extends React.Component {

  constructor(props) {
    super(props)

    this.onClick = (e) => {
      rbLog('[onClick]');
      e.preventDefault();
      const { onClick } = this.props;
      onClick(e)
    }
  }

  render() {
    const { classes, children } = this.props;
    return <div
      onMouseDown={this.onClick}
      className={classes.root}>{children}</div>;
  }
}

export const Button = injectSheet(buttonStyles)(RawButton);


export class RawMarkButton extends React.Component {
  constructor(props) {
    super(props);

    this.onToggle = (e) => {
      e.preventDefault();
      this.props.onToggle(this.props.mark);
    };
  }

  render() {
    const { classes, children, active } = this.props;
    let className = classes.button;

    if (active) {
      className += ` ${classes.active}`;
    }

    return <span
      className={className}
      onMouseDown={this.onToggle}>{children}</span>
  }
}

export const MarkButton = injectSheet(styles)(RawMarkButton);