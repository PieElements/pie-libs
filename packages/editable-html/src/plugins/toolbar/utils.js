export const findSingleNode = (value) => {

  if (!value || !value.isCollapsed || !value.startKey) {
    return;
  }

  const inline = value.document.getClosestInline(value.startKey);

  if (inline) {
    return inline;
  }

  const block = value.document.getClosestBlock(value.startKey);

  if (block) {
    return block;
  }

}