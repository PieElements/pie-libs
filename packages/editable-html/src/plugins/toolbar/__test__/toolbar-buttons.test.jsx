import { classObject, mockIconButton } from '../../../__test__/utils';

import { Data } from 'slate';
import { RawMarkButton, RawButton } from '../toolbar-buttons';
import React from 'react';
import debug from 'debug';
import renderer from 'react-test-renderer';
import { stub } from 'sinon';

mockIconButton();

const log = debug('editable-html:test:editor-and-toolbar');

describe('Button', () => {

  it('renders', () => {
    const classes = classObject('root');
    const tree = renderer.create(<RawButton classes={classes} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

});

describe('MarkButton', () => {

  const classes = classObject('button', 'root', 'active');

  it('renders not active', () => {
    const tree = renderer.create(<RawMarkButton
      active={false}
      classes={classes} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders active', () => {
    const tree = renderer.create(<RawMarkButton
      active={true}
      classes={classes} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
