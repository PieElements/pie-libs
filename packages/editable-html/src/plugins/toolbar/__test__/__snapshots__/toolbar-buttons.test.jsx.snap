// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders 1`] = `
<div
  className="root"
  onMouseDown={[Function]}
/>
`;

exports[`MarkButton renders active 1`] = `
<span
  className="button active"
  onMouseDown={[Function]}
/>
`;

exports[`MarkButton renders not active 1`] = `
<span
  className="button"
  onMouseDown={[Function]}
/>
`;
